crates/sui/tests/reconfiguration_tests.rs

```diff
@@ -436,90 +439,400 @@ async fn test_reconfig_with_committee_change_basic() {
+            (val, node_config)
+        })
+        .filter(|(val, _node)| !initial_pubkeys.contains(&val.protocol_key()))
+        .collect();
+    assert_eq!(standby_nodes.len(), 6);
+
+    let mut epoch = 0;
+
+    while !standby_nodes.is_empty() {
+        // Add 2 validators from standby_set to committee (and add to end of current_set).
+        // Remove first 2 validators from committee (and beginning of current_set) and add to end of standby_set.
+        // This effectively creates a circular buffer of two different leaving and joining
+        // validators per epoch
+
+        let joining_validators = standby_nodes.split_off(standby_nodes.len() - 2);
+        assert_eq!(joining_validators.len(), 2);
+
+        // request to add new validators
+        for (validator_info, node_config) in joining_validators.clone() {
+            let sender = node_config.sui_address();
+            let (gas_objects, stake) = object_map.get(&sender).unwrap();
+            let effects = execute_join_committee_txes(
+                &authorities,
+                gas_objects.clone(),
+                *stake,
+                &validator_info,
+                node_config,
+            )
+            .await;
+
+            let gas_objects = vec![
+                effects[0].gas_object().0,
+                effects[1].gas_object().0,
+                effects[2].gas_object().0,
+                gas_objects[3],
+            ];
+            object_map.insert(sender, (gas_objects, *stake));
+        }
+
+        // last 2 validators request to leave
+        let auth_len = authorities.len();
+        for auth in &authorities[auth_len - 2..] {
+            let node_config = validator_superset
+                .validator_configs()
+                .iter()
+                .find(|config| config.protocol_public_key() == auth.with(|node| node.state().name))
+                .unwrap();
+
+            let sender = node_config.sui_address();
+            let (gas_objects, stake) = object_map.get(&sender).unwrap();
+            let effects =
+                execute_leave_committee_txes(&authorities, gas_objects[3], node_config).await;
+
+            let gas_objects = vec![
+                gas_objects[0],
+                gas_objects[1],
+                gas_objects[2],
+                effects.gas_object().0,
+            ];
+            object_map.insert(sender, (gas_objects, *stake));
+        }
+
+        trigger_reconfiguration(&authorities).await;
+
+        // Start new Nodes
+        let mut joined_auth_handles: Vec<_> = vec![];
+        for (_val, node_config) in joining_validators.clone() {
+            // Make sure that the new validator config shares the same genesis as the initial one.
+            let mut node_config = node_config.clone();
+            node_config.genesis = initial_network.validator_configs[0].genesis.clone();
+            let handle =
+                start_node(&node_config.clone(), RegistryService::new(Registry::new())).await;
```

Comment:
If you start all nodes from beginning, then the time takes for them to join would be much more predicable?

Reply:
Done.